leakcanary-android-core/src/main/java/leakcanary/internal/activity/screen/AboutScreen.kt

```diff
@@ -53,17 +65,29 @@ internal class AboutScreen : Screen() {
           resources.getString(R.string.leak_canary_heap_dump_disabled_text),
           resources.getString(R.string.leak_canary_heap_dump_disabled_by_app)
       )
+      DISABLED_FROM_ABOUT_SCREEN -> String.format(
+            resources.getString(R.string.leak_canary_heap_dump_disabled_text),
+            resources.getString(R.string.leak_canary_heap_dump_disabled_from_ui)
+        )
       DISABLED_RUNNING_TESTS -> String.format(
           resources.getString(R.string.leak_canary_heap_dump_disabled_text),
           resources.getString(R.string.leak_canary_heap_dump_disabled_running_tests)
       )
     }
 
-  private companion object HeapDumpPolicy {
+    /**
+     * Updates leak canary config to enable/disable heap dump depending upon the Toggle switch from the about screen.
+     */
+    private fun updateConfig() {
+        LeakCanary.config = LeakCanary.config.copy(dumpHeap = heapDumpSwitchChecked)
```

Comment:
this is not the right place for this, should be done on app startup when we first compute the config.

Reply:
We want this to be done at both the places - app startup and here - so that any subsequent modifications are also applied to `LeakCanary.config`, right ?